src/extension.ts

```diff
@@ -34,15 +34,24 @@ import { existsSync, rmSync } from 'fs';
 import { CodemodConfig } from './data/codemodConfigSchema';
 import { parsePrivateCodemodsEnvelope } from './data/privateCodemodsEnvelopeSchema';
 import { SourceControlPanelProvider } from './components/webview/SourceControlPanelProvider';
-
-export const UrlParamKeys = {
-	engine: 'engine' as const,
-	beforeSnippet: 'beforeSnippet' as const,
-	afterSnippet: 'afterSnippet' as const,
-	codemodSource: 'codemodSource' as const,
-	codemodName: 'codemodName' as const,
-	codemodHashDigest: 'chd' as const,
-};
+import {
+	AlreadyLinkedError,
+	GlobalStateTokenStorage,
+	UserService,
+} from './components/userService';
+
+export const SEARCH_PARAMS_KEYS = Object.freeze({
```

Comment:
that looks like something that could be a `const enum`, am I right?

Reply:
addressed